src/application.coffee

```diff
@@ -332,6 +363,122 @@ updateStatus =
 	forceNext: false
 	intervalHandle: null
 
+updateStrategies =
+	'normal-update': (localApp, app, needsDownload, force, timeout) ->
+		Promise.try ->
+			fetch(app) if needsDownload
+		.then ->
+			Promise.using lockUpdates(localApp, force), ->
+				logSystemEvent(logTypes.updateApp, app) if localApp.imageId == app.imageId
+				selectAndKill(localApp.appId)
+				.then ->
+					start(app)
+			.catch (err) ->
+				logSystemEvent(logTypes.updateAppError, app, err) unless err.isLocked?
+				throw err
+	'kill-before-download': (localApp, app, needsDownload, force, timeout) ->
```

Comment:
It feels like the signature for these functions might be better as an options object, ie `({ localApp, app, needsDownload, force, timeout })`